src/GraphQL.Tests/Types/EnumGraphTypeTests.cs

```diff
@@ -123,5 +124,38 @@ public void does_not_allow_nulls_to_be_added()
         {
             Assert.Throws<ArgumentNullException>(() => new EnumerationGraphType().AddValue(null));
```

Comment:
@Shane32 Check this test case, please. If I understand correctly now we allow to have `null` as internal value for enums.

Reply:
Side note 1:
```csharp
 protected override string ChangeEnumCase(string val)
            {
                return new string(val.Select(c => char.IsUpper(c) ? char.ToLower(c) : char.ToUpper(c)).ToArray());
            }
```
->
```csharp
 protected override string ChangeEnumCase(string val) => val.ToUpper();
```

Reply:
Side note 2, new test:
```csharp
        [Fact]
        public void parses_from_value()
        {
            type.ParseValue(Colors.Red).ShouldBe(Colors.Red);
        }
```

Reply:
> null as internal value

ok we could add that feature.

> change enum case

That code is designed to allow tests for custom cased enumeration members.  But the test class is completely unused.  I don't know why it's there.  Do you want me to delete it?  Certainly changing it to be uppercase defeats the purpose of the class.

> new test

parses_from_value would fail.  Externally it is represented as a string.  Does this need to be changed?

Reply:
> Do you want me to delete it?

Yes.

>  Certainly changing it to be uppercase defeats the purpose of the class.

Aaaaa :) 🤦‍♂️

Reply:
>  Externally it is represented as a string. Does this need to be changed?

I think that externally enum **can** be represented as enum. See `GuidGraphType.ParseValue` or `UriGraphType.ParseValue`. Both accept their `native` types. Why not to do so with enums?

Reply:
I do not think Guid or Uri should allow Guid or Uri externally.

Reply:
At least with Guid or Uri there is no possibility of ambiguity.  Here there is.

Reply:
Imagine hand-crafted variables (inputs). After all, the specification does not say that variables should be formed only in the process of deserialization. Variables are simply coming from "somewhere". For example, you can read from the file or receive from other program/method. The variables can be already of necessary (internal) types. The specification says that variables can be of some runtime-type. In general, I think it cannot clearly hold the border between the external representation and internal. It all depends on who and how these variables creates. That is why the specification leaves a wider space for options compared to literals. So IMO it is **absolutely** normal to accept Guid, Uri, Enum or some other .NET types in `ParseValue` methods.

Reply:
Did you delete class?

Reply:
No I forgot.

Reply:
Done